docs/designs/015-custom-drain-extensibility.md

```diff
@@ -0,0 +1,237 @@
+    B --> C{Custom Drain Enabled?}
+    C -->|Yes| D[Load Template from ConfigMap]
+    C -->|No| E[Standard Drain Mode]
+    D --> F[Render Template with Event Data]
+    F --> G[Create Customer CR via Dynamic Client]
+    G --> H[Watch CR Status via Informer]
+    H --> I{Status Phase?}
+    I -->|Complete| J[Proceed with Pod Eviction]
+    I -->|Failed| K[Error/Retry]
+    I -->|Pending/InProgress| H
+```
+
+### Template Variables
+
+```go
+type TemplateData struct {
+  HealthEvent   HealthEevent
+  PodsToDrain   map[string][]string
+}
+```
+
+### Configuration (TOML)
+
+```toml
+[customDrain]
+    enabled = false
+    templateMountPath = "/etc/drain-templates"
+    templateFileName = "drain-template.yaml"
+    namespace = "nvsentinel"
+    apiGroup = "nvsentinel.nvidia.com"
+    version = "v1alpha1"
+    kind = "SlurmDrainRequest"
+
+    timeout = "3600s"
+
+    # Status watching (follows Kubernetes standard: InProgress, Complete, Failed)
+    statusConditionType = "Complete"
+    statusConditionStatus = "True"
+```
+
+### Template Example
+
+```yaml
+apiVersion: drain.example.com/v1alpha1
+kind: DrainRequest
+metadata:
+    name: drain-{{ .HealthEvent.NodeName }}-{{ .HealthEvent.EventID }}
+    namespace: nvsentinel
+spec:
+    nodeName: {{ .HealthEvent.NodeName }}
+    checkName: {{ .HealthEvent.CheckName }}
+    recommendedAction: {{ .HealthEvent.RecommendedAction }}
+    errorCode: {{ .HealthEvent.ErrorCode }}
+
+    entitiesImpacted:
+    {{- range .HealthEvent.EntitiesImpacted }}
+    - type: {{ .EntityType }}
+      value: {{ .EntityValue }}
+    {{- end }}
+
+    metadata:
+    {{- range $key, $value := .HealthEvent.Metadata }}
+        {{ $key }}: {{ $value }}
+    {{- end }}
+
+    reason: "{{ .HealthEvent.Message }}"
+
+    podsToDrain:
+    {{- range $namespace, $pods := .PodsToDrain }}
+        {{ $namespace }}:
+        {{- range $pods }}
+            - {{ . }}
+        {{- end }}
+    {{- end }}
+status:
+  conditions:
+    - type: Complete
+      status: "False"
+      reason: Pending
+```
+
+### Node-Drainer Changes
+
+**1. Evaluator** (`evaluator/evaluator.go`):
+```go
+// Check if custom drain enabled
+if e.config.CustomDrain.Enabled {
+    return &DrainActionResult{
+        Action:  ActionCreateCR,
+        Timeout: e.config.CustomDrain.Timeout.Duration,
+    }, nil
+}
+```
+
+**2. Custom Drain Client** (`customdrain/client.go`):
+```go
+// Same pattern as fault-remediation
+func (c *Client) CreateDrainCR(ctx context.Context, data TemplateData) (string, error) {
+    // 1. Execute template
+    // 2. Parse YAML to unstructured
+    // 3. Create CR via dynamic client
+    // 4. Return CR name
+}
+
+func (c *Client) GetCRStatus(ctx context.Context, crName string) (bool, error) {
+    // 1. Get CR via dynamic client
+    // 2. Extract status.conditions array
+    // 3. Check if condition type matches config and status is True
+    // 4. Return true if complete, false otherwise
```

Comment:
Once the CR is deemed to be complete, should we delete it similar to how the `TokenRequest` works? This will prevent an unnecessary build-up of CRs